proxy-identity/main.go

```diff
@@ -124,11 +127,14 @@ func generateAndStoreKey(p string) (key *ecdsa.PrivateKey, err error) {
 }
 
 func generateAndStoreCSR(p, id string, key *ecdsa.PrivateKey) ([]byte, error) {
-	// TODO do proper DNS name validation.
 	if id == "" {
 		return nil, errors.New("a non-empty identity is required")
 	}
 
+	if err := validation.IsFullyQualifiedDomainName(field.NewPath(""), id).ToAggregate(); err != nil {
+		return nil, errors.New("a qualified DNS identity is required")
```

Comment:
let's put the id in the error message to make this error more helpful.

Reply:
updated